http/src/client/interaction.rs

```diff
@@ -167,16 +167,13 @@ impl<'a> InteractionClient<'a> {
     /// the same guild will overwrite the old command. See [the discord docs]
     /// for more information.
     ///
-    /// Returns an error of type [`NameInvalid`] error type if the command name
-    /// is not between 1 and 32 characters.
-    ///
     /// [`NameInvalid`]: twilight_validate::command::CommandValidationErrorType::NameInvalid
     /// [the discord docs]: https://discord.com/developers/docs/interactions/application-commands#create-guild-application-command
-    pub fn create_guild_command(
+    pub const fn create_guild_command(
         &'a self,
         guild_id: Id<GuildMarker>,
         name: &'a str,
```

Comment:
since is this targeting next, i'd like you to remove the `name` parameter here. it will make it easier when we eventually also accept a built `command` with the upcoming `payload` feature. it also creates more of a logical difference between error types returned by the different validation done on names.